borg/key.py

```diff
@@ -396,19 +400,30 @@ class KeyfileKey(KeyfileKeyBase):
     TYPE = 0x00
     FILE_ID = 'BORG_KEY'
 
+    def sanity_check(self, filename):
+        with open(filename, 'r') as fd:
+            line = fd.readline().strip()
+            if line.startswith(self.FILE_ID) and line[len(self.FILE_ID) + 1:] == id:
+                return filename
+            else:
+                return None
```

Comment:
how about:
if not line.startswith(self.FILE_ID):
    raise KeyfileInvalid
if line[len(self.FILE_ID) + 1:] != id:
    raise KeyfileMismatch
return filename